File: Templates/Java/integrationtesting/src/main/java/com/example/integrationtesting/DistanceUtils.java
package com.example.integrationtesting;

public class DistanceUtils {
    private DistanceUtils() {}

    public static double calculateDistance(double lat1, double lon1, double lat2, double lon2) {
        // Validate latitude and longitude ranges
        if (lat1 < -90 || lat1 > 90 || lat2 < -90 || lat2 > 90 ||
            lon1 < -180 || lon1 > 180 || lon2 < -180 || lon2 > 180) {
            throw new IllegalArgumentException("Latitude must be between -90 and 90, longitude between -180 and 180.");
        }

        double R = 3963.19; // Earth's radius in miles
        double dLat = toRadians(lat2 - lat1);
        double dLon = toRadians(lon2 - lon1);

        double a = Math.pow(Math.sin(dLat / 2), 2) +
                   Math.cos(toRadians(lat1)) * Math.cos(toRadians(lat2)) *
                   Math.pow(Math.sin(dLon / 2), 2);

        double c = 2 * Math.atan2(Math.sqrt(a), Math.sqrt(1 - a));
        return R * c; // Distance in miles
    }

    private static double toRadians(double degrees) { 
        return degrees * Math.PI / 180.0;
    }
}

File: Templates/Kotlin/integrationtesting/src/main/java/com/example/integrationtesting/DistanceUtils.kt
package com.example.integrationtesting

object DistanceUtils {
    @JvmStatic
    fun calculateDistance(lat1: Double, lon1: Double, lat2: Double, lon2: Double): Double {
        val earthRadiusMiles = 3958.8

        val dLat = Math.toRadians(lat2 - lat1)
        val dLon = Math.toRadians(lon2 - lon1)

        val originLat = Math.toRadians(lat1)
        val destinationLat = Math.toRadians(lat2)

        val a = Math.sin(dLat / 2) * Math.sin(dLat / 2) +
        Math.cos(originLat) * Math.cos(destinationLat) *
        Math.sin(dLon / 2) * Math.sin(dLon / 2)

        val c = 2 * Math.asin(Math.sqrt(a))

        return earthRadiusMiles * c
    }
}
